packages/core/src/external/aws/s3.ts

```diff
@@ -85,4 +97,41 @@ export class S3Utils {
       Expires: durationSeconds ?? DEFAULT_SIGNED_URL_DURATION,
     });
   }
+
+  async getPresignedUrl({
+    bucket,
+    key,
+    durationSeconds,
+  }: {
+    bucket: string;
+    key: string;
+    durationSeconds?: number;
+  }): Promise<string> {
+    const command = new PutObjectCommand({ Bucket: bucket, Key: key });
+    const presignedUrl = await getPresignedUrl(this.s3Client, command, {
+      expiresIn: durationSeconds ?? DEFAULT_SIGNED_URL_DURATION,
+    });
+    return presignedUrl;
+  }
+
+  parseS3FileName(fileKey: string): { cxId: string; patientId: string; docId: string } | undefined {
+    if (fileKey.includes("_")) {
+      const keyParts = fileKey.split("_");
+      if (keyParts[0] && keyParts[1] && keyParts[2] && keyParts[0].includes("/")) {
+        const cxIdParts = keyParts[0].split("/");
+        if (cxIdParts[0]) {
+          const cxId = cxIdParts[0];
+          const patientId = keyParts[1];
+          const docId = keyParts[2];
+          const fileData = {
+            cxId,
+            patientId,
+            docId,
+          };
+          return fileData;
+        }
+      }
+    }
+    return;
+  }
```

Comment:
ditto creating consts for the parts make it easier to read/update.

could also move this to be right below `createS3FileName` to maximize the benefit of being in one place 😄